__tests__/todoList.spec.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import { userDataBuilder, itemDataBuilder } from 'dataBuilders';
+import { submitInput, submitMultiples } from 'submitInput';
+
+import Todo from '../src/screens/todo';
+
+const user = userDataBuilder();
+const todoItem = itemDataBuilder();
+
+test('all item should be toggle at the same time', () => {
+  //* render the todo screen
+  const { getByPlaceholderText, getByText, getByLabelText, getAllByRole } =
+    render(<Todo user={user} />);
+
+  //* get to-do list, items counter and input
+  const counter = getByText(/active: */i);
+  const input = getByPlaceholderText(/What needs to be done?/i);
+
+  //* type in the input field & click submits multiple times
+  submitMultiples(input, todoItem.extraWords);
+
+  //* Press toggle-all button
+  fireEvent.press(getByLabelText('complete-all-button'));
+
+  //* get one switch for each to-do item
+  const switchArray = getAllByRole('switch');
+
+  //* to-do list counter should decrease to zero
+  expect(counter).toHaveTextContent(/ *0/i);
+
+  //* check all switches with words
+  switchArray.forEach((switchElement) => {
+    //* value prop should be true
+    expect(switchElement).toHaveProp('value', true);
```

Comment:
Why are we falling into this again?

Reply:
My logic here: this test is to see if `complete-all-button` toggle all item when press -> to see if they are toggle i need to check the `value` prop -> i could check the visible changes of the item but thats another test in another file already

What matters here is if the button works and not so much if the item change when toggle, `todoItem.spec.js` check if the item work as expected and `todoList.spec.js` check if the list as a whole work as expected

Also i thought this: if by any reason the item behavior when toggle change or is refactor, the `complete-all-button` can (and most likely will) still work then the test that should fail should be in `todoItem.spec.js` not `todoList.spec.js`

Reply:
Mmm... I have encountered feeling but I like you have a reason for doing this. Let's roll like this 👌